### PR TITLE
feat(icon): add icon function/helper

### PR DESCRIPTION
**Pull request by cal-smith**
This allows us to re-use the existing `@carbon/icons` work and conventions.

### CODE REVIEW

**Comment by cal-smith** [comment]
Added a few more demos, and included support for properly accessible `title`s ... before this is ready I'd like to submit my changes to `toString` back to `icon-helpers`

**Comment by cal-smith** [comment]
Yeah `@carbon/icons-lit` definitely makes sense, and I think that will be the final version of this icon component/toolkit.

That said, I'm not sure that the overhead of converting the icons to string at runtime would have that much of a performance impact, especially if we add memoization ... we'd definitely need to quantify it.

File: src/components/icon/icon.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * @param {object} attributes Optional. object representation of SVG attributes
+ */
+export const icon = (descriptor, attributes = {}) => {
+  descriptor.attrs = getAttributes(Object.assign(descriptor.attrs, attributes));
+  if (descriptor.attrs.title) {
+    const id = `__carbon-icon__${Math.random()
+      .toString(36)
+      .substr(2)}`;
+    descriptor.content.push({
+      elem: 'title',
```
[the remark this answers is not in the document]
Comment (cal-smith) [reply src/components/icon/icon.js]: For sure

**Comment by asudoh** [comment]
Thank you for sharing your thoughts @cal-smith! I personally haven't seen a Carbon application with large number of icons that would make run-time descriptor conversion a performance concern, and thus current solution is fine as long as we ultimately are going with `@carbon/icons-lit`.

**Comment by asudoh** [comment]
@cal-smith Would you update `yarn.lock`? Thanks!

**Comment by asudoh** [comment]
@cal-smith Cool thank you for the update! Would you want to go forward with the current code, or do you want to still change `icon-helpers`?

**Comment by cal-smith** [comment]
We can go ahead as is :+1: I haven't had a chance to do as much work on this as I would have liked lately, so we can merge this for now

**Pull request merged**